File: HackerRank 30 Days/day14/main.cs
using System;
using System.Linq;

class Difference
{
    private int[] elements;
    public int maximumDifference;

    public Difference(int[] a)
    {
        elements = a;
    }

    public void computeDifference()
    {
        int max = 0;
        for (int i = 0; i < elements.Length; i++)
        {
            for (int j = i + 1; j < elements.Length; j++)
            {
                int diff = Math.Abs(elements[i] - elements[j]);
                if (diff > max)
                {
                    max = diff;
                }
            }
        }
        maximumDifference = max;
    }
}

class Solution
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int[] a = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

        Difference d = new Difference(a);
        d.computeDifference();
        Console.WriteLine(d.maximumDifference);
    }
}

File: HackerRank 30 Days/day14/main.js
class Difference {
    constructor(a) {
        this.elements = a;
    }

    computeDifference() {
        let maximumDifference = 0;
        for (let i = 0; i < this.elements.length; i++) {
            for (let j = i + 1; j < this.elements.length; j++) {
                const diff = Math.abs(this.elements[i] - this.elements[j]);
                if (diff > maximumDifference) {
                    maximumDifference = diff;
                }
            }
        }
        this.maximumDifference = maximumDifference;
    }
}

const n = parseInt(readLine(), 10);
const a = readLine().split(' ').map(Number);

const d = new Difference(a);
d.computeDifference();

console.log(d.maximumDifference);
